src/bundle/ui-dev/src/modules/multi-file-upload/components/upload-list/upload.item.component.js

```diff
@@ -561,26 +572,43 @@ export default class UploadItemComponent extends Component {
     }
 
     render() {
-        if (this.state.deleted) {
+        const {
+            uploaded,
+            aborted,
+            disallowedType,
+            disallowedSize,
+            failed,
+            uploading,
+            disallowedContentType,
+            deleted,
+            totalSize,
+        } = this.state;
+        const isError = !uploaded && !aborted && (disallowedSize || disallowedType || disallowedContentType) && failed && !uploading;
+        const wrapperClassName = createCssClassNames({
+            'c-upload-list-item': true,
+            'c-upload-list-item--errored': isError,
+        });
+
+        if (deleted) {
             return null;
         }
 
         return (
-            <div className="c-upload-list-item">
-                <div className="c-upload-list-item__icon-wrapper">{this.renderIcon()}</div>
+            <div className={wrapperClassName}>
+                <div className="c-upload-list-item__icon-wrapper">{!isError && this.renderIcon()}</div>
                 <div className="c-upload-list-item__meta">
                     <div className="c-upload-list-item__name">{this.props.data.file.name}</div>
-                    <div className="c-upload-list-item__size">{this.state.uploaded ? this.state.totalSize : ''}</div>
+                    <div className="c-upload-list-item__size">{!isError && uploaded ? totalSize : ''}</div>
```

Comment:
Nitpicking: Is this `uploaded` here even needed? `isError` checks already value of upladed variable

Reply:
totalSize should not be visible during upload.